src/libs/ActiveClientManager/index.ts

```diff
@@ -70,19 +72,37 @@ const isClientTheLeader: IsClientTheLeader = () => {
 const cleanUpClientId = () => {
     isPromotingNewLeader = isClientTheLeader();
     activeClients = activeClients.filter((id) => id !== clientID);
-    ActiveClients.setActiveClients(activeClients);
+    setActiveClients(activeClients);
+};
+
+const removeBeforeUnloadListener = () => {
+    if (!beforeunloadListenerAdded) {
+        return;
+    }
+    beforeunloadListenerAdded = false;
+    window.removeEventListener('beforeunload', cleanUpClientId);
 };
 
 /**
  * Add our client ID to the list of active IDs.
  * We want to ensure we have no duplicates and that the activeClient gets added at the end of the array (see isClientTheLeader)
  */
 const init: Init = () => {
+    removeBeforeUnloadListener();
     activeClients = activeClients.filter((id) => id !== clientID);
     activeClients.push(clientID);
-    ActiveClients.setActiveClients(activeClients).then(resolveSavedSelfPromise);
+    setActiveClients(activeClients).then(resolveSavedSelfPromise);
 
-    window.addEventListener('beforeunload', cleanUpClientId);
+    beforeunloadListenerAdded = true;
+    window.addEventListener('beforeunload', () => {
+        // When we open route in desktop, beforeunload is fired unexpectedly here.
+        // So we should return early in this case to prevent cleaning the clientID
+        if (isOpeningRouteInDesktop()) {
+            resetIsOpeningRouteInDesktop();
+            return;
+        }
+        cleanUpClientId();
+    });
```

Comment:
The `addEventListener` is adding a new function everytime, and the `removeEventListener` won't remove the function given that `cleanUpClientId` doesn't exists. Probably that's why you need an extra check. 
The `Add/RemoveEventListener` must receive the same function instance.

Reply:
It depends, if doing 
```
window.addEventListener('beforeunload', cleanUpClientId);
```
desktop doesn't fire `beforeunload`, probably is not needed, but if just to really make sure, maybe yes, can be inside of `cleanUpClientId`